src/libraries/System.Private.CoreLib/src/System/IO/FileSystem.TryCloneFile.OSX.cs

```diff
@@ -0,0 +1,87 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Win32.SafeHandles;
+using System.Diagnostics;
+
+namespace System.IO
+{
+    internal static partial class FileSystem
+    {
+        private static bool TryCloneFile(string sourceFullPath, string destFullPath, bool overwrite)
+        {
+            // Try to clone the file immediately, this will only succeed if the
+            // destination doesn't exist, so we don't worry about locking for this one.
+            int flags = Interop.@libc.CLONE_ACL;
+            retry1:
+            if (Interop.@libc.clonefile(sourceFullPath, destFullPath, flags) == 0)
```

Comment:
Can you add a static method in this method:

```
static bool TryCloneFile(string sourceFullPath, string destFullPath, bool overwrite, out Interop.Error error)
```

and call that two times instead of the labels and gotos?

Reply:
I think I implemented what you meant, let me know if I did it wrong.